gobblin-api/src/main/java/gobblin/configuration/WorkUnitState.java

```diff
@@ -171,14 +186,16 @@ public State getPreviousTableState() {
   @Override
   public void readFields(DataInput in)
       throws IOException {
-    workunit.readFields(in);
+    this.workunit.readFields(in);
+    this.watermarkInterval.readFields(in);
```

Comment:
Yeah. Since we probably want to avoid another state-store migration, I added a hack to ensure the serialization still works, but no state-store migration is necessary.